process/api.py

```diff
@@ -87,6 +99,161 @@ class CollectionViewSet(viewsets.ViewSetMixin, ListAPIView):
+        if compiled_collection:
+            result["compiled_collection_id"] = compiled_collection.pk
+
+        return Response(result)
+
+    @csrf_exempt
+    @action(methods=["post"], detail=False)
+    def close(self, request):
+        input_message = request.data
+
+        if "collection_id" not in input_message:
+            return Response(
+                'Unable to parse input. Please provide {"collection_id": "<collection_id>"}',
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        with transaction.atomic():
+            collection = Collection.objects.select_for_update().get(pk=input_message["collection_id"])
+            if input_message.get("stats"):
+                # this value is used later on to detect, whether all collection has been processed yet
+                collection.expected_files_count = input_message["stats"].get(
+                    "kingfisher_process_expected_files_count", 0
+                )
+            collection.store_end_at = Now()
+            collection.save(update_fields=["expected_files_count", "store_end_at"])
+
+            upgraded_collection = collection.get_upgraded_collection()
+            if upgraded_collection:
+                upgraded_collection.expected_files_count = collection.expected_files_count
+                upgraded_collection.store_end_at = Now()
+                upgraded_collection.save(update_fields=["expected_files_count", "store_end_at"])
+
+            if input_message.get("reason"):
+                create_note(collection, CollectionNote.Level.INFO, f"Spider close reason: {input_message['reason']}")
+                if upgraded_collection:
+                    create_note(
+                        upgraded_collection,
+                        CollectionNote.Level.INFO,
+                        f"Spider close reason: {input_message['reason']}",
+                    )
+
+            if input_message.get("stats"):
+                create_note(collection, CollectionNote.Level.INFO, "Spider stats", data=input_message["stats"])
+                if upgraded_collection:
+                    create_note(
+                        upgraded_collection, CollectionNote.Level.INFO, "Spider stats", data=input_message["stats"]
+                    )
+
+        with get_publisher() as client:
+            message = {"collection_id": collection.pk}
+            client.publish(message, routing_key="collection_closed")
+
+            if upgraded_collection:
+                message = {"collection_id": upgraded_collection.pk}
+                client.publish(message, routing_key="collection_closed")
+
+            if compiled_collection := collection.get_compiled_collection():
+                message = {"collection_id": compiled_collection.pk}
+                client.publish(message, routing_key="collection_closed")
+
+        return Response(status=status.HTTP_202_ACCEPTED)
+
+    @csrf_exempt
+    @action(methods=["post"], detail=False)
+    def wipe(self, request):
```

Comment:
Change to `def destroy`, which has special meaning (and doesn't require `@action`), when used with correct superclass. The API then becomes `DELETE /collections/{id}/` and we won't need to parse the request body.